Simple user auth

coffeestrike opened the pull request:
1. Included pundit
- created survey policy

2. Survey controller
- before_action :is_signed_in?

3. Application controller
- def is_signed_in?

4. Migrations
- removed User.role column (string)
- added User.role column (integer - enum)

Only a logged in person can create a survey

coffeestrike committed:
Simple user auth

1. Included pundit
- created survey policy

2. Survey controller
- before_action :is_signed_in?

3. Application controller
- def is_signed_in?

4. Migrations
- removed User.role column (string)
- added User.role column (integer - enum)

Only a logged in person can create a survey

pynkvalerian commented on app/controllers/application_controller.rb:
```diff
@@ -1,4 +1,13 @@
   protect_from_forgery with: :exception
+
+  def is_signed_in?
+    if !signed_in?
```
`unless signed_in?` might be more readable

pynkvalerian commented on app/controllers/users_controller.rb:
```diff
@@ -0,0 +1,47 @@
+    @user = User.new(user_from_params)
+    if @user.save
+      sign_in @user
+      redirect_back_or url_after_create
```
oh nice! I didn't know about that clearance provides this `url_after_create` !

pynkvalerian commented on app/policies/application_policy.rb:
```diff
@@ -0,0 +1,53 @@
+      scope
+    end
+  end
+end
```
If i'm not mistaken, this is just the example given by pundit, can be deleted :)

pynkvalerian commented on config/routes.rb:
```diff
@@ -1,4 +1,11 @@
+  resource :session, controller: "clearance/sessions", only: [:create]
+
+  resources :users, controller: "users", only: [:create, :index, :show, :edit, :update] do
+    resource :password, controller: "clearance/passwords", only: [:create, :edit, :update]
```
you have `resources :passwords` AND `resource :password`, which is in use? :)

coffeestrike committed:
Made changes requested & installed clearance:spec

coffeestrike committed:
Simple spec for survey new

Clearance
- Included Clearance:spec (for spec helpers)
- removed some unused Clearance:specs

Spec
- created user & admin factory
- redirects to home when Pundit::NotAuthorizedError
- disabled key_value_pairs spec & model validation since unimplemented

Pundit
- removed redundant authorizations since using *before_action method

pynkvalerian commented on app/controllers/surveys_controller.rb:
```diff
@@ -23,10 +28,12 @@ def create
 
   def update
     @survey = Survey.find(params[:id])
+    byebug
```
byebug

pynkvalerian commented on app/controllers/surveys_controller.rb:
```diff
@@ -8,12 +11,14 @@ def show
 
   def create
     @survey = Survey.new(survey_params)
+    @survey.user_id = current_user.id
```
could be written as `@survey = current_user.surveys.new(survey_params)`

coffeestrike committed:
Changes as requested

- simplified @survey creation
- removed byebug in SurveyController update method

coffeestrike committed:
removed byebug
